### PR TITLE
Fix for broken console functions in examples

### PR DESCRIPTION
**Pull request by m-cheung**

### CODE REVIEW

File: test/browser/FOAMByExampleRunner.js [hunk 1]
```diff
@@ -191,4 +191,14 @@ foam.async.repeat(FBE.length, function runExemplar(index) {
     writeOutput();
     return result;
   }
-})();
+})().then(function() {
+  // reverting logging strategy performed during async calls
+  console = {
+    __proto__: oldConsole,
```
Comment (mdittmer) [inline test/browser/FOAMByExampleRunner.js]: This looks right, although we don't need to override `oldConsole`'s implementations with `oldConsole`'s implementations. `console = oldConsole` should suffice.

**Review by mdittmer — CHANGES_REQUESTED** [review]
Got the async/promisey pattern right, but we don't need the extra overrides.

File: test/browser/FOAMByExampleRunner.js (hunk 1 above)
Comment (m-cheung) [inline test/browser/FOAMByExampleRunner.js]: The change has been made as requested

**Review by m-cheung — COMMENTED** [review]
The change has been made as requested

**Review by mdittmer — APPROVED** [review]
LGTM

**Pull request merged**